cg/store/crud/read.py

```diff
@@ -380,7 +380,7 @@ def get_illumina_metrics_entry_by_device_sample_and_lane(
 
     def get_illumina_sequencing_run_by_device_internal_id(
         self, device_internal_id: str
-    ) -> IlluminaSequencingRun:
+    ) -> IlluminaSequencingRun | None:
```

Comment:
add error handling if sequencing run not found.

Reply:
Should that be here or in the calling function though? I think it is probably up to the caller to decide on what to do with that info?

Reply:
I think for these basic data access functions where we provide a unique identifier, it makes sense to raise an exception - it is really weird if we don't get a hit for the identifier.

Reply:
I might have written some `is_entry_in_status_db` methods which are basically wrapping the returns of these methods with a bool... Probably bad practice but something that needs to be looked at since they will crash with the try except change.

Reply:
Fixed in separate PR